### DISCUSSION TITLE
Sending text to specific OS kitty window

### DISCUSSION THREAD

**Discussion by Tokariew** [kovidgoyal/kitty#3762] [Q&A] [answered]
How to send text to kitty window (not build in kitty, but OS one `new_os_window`) running specific program from systemd service?
Allowing remove control and setting listen_on seems like way to go, but with multiple os windows, i don't know how to setup it…
On other terminals i would use xdotool, but on kitty, issue #2261 prevent my service from working correctly.

**Comment by kovidgoyal** [comment, marked as answer]
yes, use remote control, which allows you to select windows by various criteria including their titles, ids, commands running in them, etc. *Far* more powerful than xdotool

**Reply by Tokariew** [reply]
it seems that with remote control I can see activity from other kitty windows spawn with `new_os_window` and not from my Window Manager…
So how should i setup listen_on to be able to easily iterate from all listening kitty windows, even spawn from my Window Manager?

**Reply by kovidgoyal** [reply]
Change your window manager to run kitty as kitty --single-instance
